File: mobile/lib/models/food_provider.dart
class FoodProvider {
  final int id;
  final int userId;
  final String providerType;
  final String businessName;
  final String? description;
  final String? logoUrl;
  final String? phone;
  final String? email;
  final String? address;
  final double? latitude;
  final double? longitude;
  final String? city;
  final double avgRating;
  final int totalOrders;
  final bool isFeatured;
  final bool isActive;
  final String verificationStatus;

  const FoodProvider({
    required this.id,
    required this.userId,
    required this.providerType,
    required this.businessName,
    this.description,
    this.logoUrl,
    this.phone,
    this.email,
    this.address,
    this.latitude,
    this.longitude,
    this.city,
    this.avgRating = 0.0,
    this.totalOrders = 0,
    this.isFeatured = false,
    this.isActive = true,
    this.verificationStatus = 'pending',
  });

  factory FoodProvider.fromJson(Map<String, dynamic> json) {
    return FoodProvider(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      providerType: json['provider_type'] as String? ?? 'kitchen',
      businessName: json['business_name'] as String? ?? '',
      description: json['description'] as String?,
      logoUrl: json['logo_url'] as String?,
      phone: json['phone'] as String?,
      email: json['email'] as String?,
      address: json['address'] as String?,
      latitude: json['latitude'] != null
          ? (json['latitude'] as num).toDouble()
          : null,
      longitude: json['longitude'] != null
          ? (json['longitude'] as num).toDouble()
          : null,
      city: json['city'] as String?,
      avgRating: (json['avg_rating'] as num? ?? 0).toDouble(),
      totalOrders: json['total_orders'] as int? ?? 0,
      isFeatured: json['is_featured'] as bool? ?? false,
      isActive: json['is_active'] as bool? ?? true,
      verificationStatus: json['verification_status'] as String? ?? 'pending',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'provider_type': providerType,
      'business_name': businessName,
      'description': description,
      'logo_url': logoUrl,
      'phone': phone,
      'email': email,
      'address': address,
      'latitude': latitude,
      'longitude': longitude,
      'city': city,
      'avg_rating': avgRating,
      'total_orders': totalOrders,
      'is_featured': isFeatured,
      'is_active': isActive,
      'verification_status': verificationStatus,
    };
  }
}

File: app/Models/Food/FoodProvider.php
<?php

namespace App\Models\Food;

use App\Models\User;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class FoodProvider extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'provider_type',
        'business_name',
        'description',
        'logo_url',
        'cover_image_url',
        'phone',
        'email',
        'address',
        'latitude',
        'longitude',
        'city',
        'state',
        'pincode',
        'fssai_license',
        'fssai_expiry',
        'gst_number',
        'pan_number',
        'verification_status',
        'verified_at',
        'is_active',
        'is_featured',
        'avg_rating',
        'total_orders',
        'total_revenue',
        'commission_rate',
        'operating_hours',
        'delivery_radius_km',
        'min_order_amount',
        'free_delivery_above',
        'bank_account_number',
        'bank_ifsc',
        'upi_id',
    ];

    protected $hidden = [
        'bank_account_number',
        'bank_ifsc',
        'upi_id',
    ];

    protected $casts = [
        'latitude' => 'decimal:7',
        'longitude' => 'decimal:7',
        'fssai_expiry' => 'date',
        'verified_at' => 'datetime',
        'is_active' => 'boolean',
        'is_featured' => 'boolean',
        'avg_rating' => 'decimal:2',
        'total_orders' => 'integer',
        'total_revenue' => 'decimal:2',
        'commission_rate' => 'decimal:2',
        'operating_hours' => 'array',
        'delivery_radius_km' => 'integer',
        'min_order_amount' => 'decimal:2',
        'free_delivery_above' => 'decimal:2',
    ];

    // ── Relationships ────────────────────────────────────────────────

    /**
     * Get the user who owns this provider profile.
     */
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    /**
     * Get the food items offered by this provider.
     */
    public function items(): HasMany
    {
        return $this->hasMany(FoodItem::class, 'provider_id');
    }

    /**
     * Get the delivery slots offered by this provider.
     */
    public function deliverySlots(): HasMany
    {
        return $this->hasMany(FoodDeliverySlot::class, 'provider_id');
    }

    /**
     * Get the orders placed with this provider.
     */
    public function orders(): HasMany
    {
        return $this->hasMany(FoodOrder::class, 'provider_id');
    }

    /**
     * Get the reviews for this provider.
     */
    public function reviews(): HasMany
    {
        return $this->hasMany(FoodReview::class, 'provider_id');
    }

    /**
     * Get the catering quotes sent by this provider.
     */
    public function cateringQuotes(): HasMany
    {
        return $this->hasMany(CateringQuote::class, 'provider_id');
    }

    /**
     * Get the hotel food services for this provider.
     */
    public function hotelServices(): HasMany
    {
        return $this->hasMany(HotelFoodService::class, 'provider_id');
    }

    /**
     * Get the promotions offered by this provider.
     */
    public function promotions(): HasMany
    {
        return $this->hasMany(FoodPromotion::class, 'provider_id');
    }

    // ── Scopes ───────────────────────────────────────────────────────

    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }

    public function scopeVerified($query)
    {
        return $query->where('verification_status', 'approved');
    }

    public function scopeFeatured($query)
    {
        return $query->where('is_featured', true);
    }

    public function scopePendingVerification($query)
    {
        return $query->where('verification_status', 'pending');
    }

    public function scopeByType($query, string $type)
    {
        return $query->where('provider_type', $type);
    }

    // ── Helpers ──────────────────────────────────────────────────────

    public function isVerified(): bool
    {
        return $this->verification_status === 'approved';
    }

    public function isActive(): bool
    {
        return $this->is_active === true;
    }
}
